src/marinade-state/marinade-state.ts

```diff
@@ -168,18 +173,21 @@ export class MarinadeState {
     })
   }
 
-  async getStakeInfos(): Promise<StakeInfo[]> {
+  /**
+   * return listStakeInfos with capacity
+   */
+  async getStakeInfos(): Promise<[StakeInfo[], number]> {
     const stakeRecords = await this.getStakeRecords()
```

Comment:
destructure here `const {stakeRecords, capacity} = await this.getStakeRecords()`

Reply:
I've updated it. please have a look.